contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -731,19 +750,36 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
             return;
         }
 
-        // Call the destination address of the message with the formatted call data.
-        // Only provide the required gas limit to the sub-call so that the end application
-        // cannot consume an arbitrary amount of gas.
-        // solhint-disable-next-line avoid-low-level-calls
-        (bool success, ) = message.destinationAddress.call{
-            gas: message.requiredGasLimit
-        }(
-            abi.encodeCall(
-                ITeleporterReceiver.receiveTeleporterMessage,
-                (originChainID, message.senderAddress, message.message)
-            )
+        // Encode the payload by ABI encoding a call to the {receiveTeleporterMessage} function
+        // defined by the {ITeleporterReceiver} interface.
+        bytes memory payload = abi.encodeCall(
+            ITeleporterReceiver.receiveTeleporterMessage,
+            (originChainID, message.senderAddress, message.message)
         );
 
+        // Call the destination address of the message with the formatted call data. Only provide the required
+        // gas limit to the sub-call so that the end application cannot consume an arbitrary amount of gas.
+        //
+        // Assembly is used for the low-level call to avoid unnecessary memory expansion of the return data.
+        // This prevents possible "return bomb" vectors where the external contract could force the caller
+        // to use an arbitrary amount of gas. First need to define the {target}, {requiredGas}, and {success}
+        // variables to make them easily accesible within and follow the assembly block.
+        address target = message.destinationAddress;
+        uint256 requiredGas = message.requiredGasLimit;
+        bool success;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
```

Comment:
the two assembly calls are pretty similar, any chance we can abstract to a function and use for both? 

Also, any unit tests for the assembly call that can be added?

Reply:
Good call on moving the assembly to a helper 👍 

Our unit tests actually already cover the assembly usage. The expected behavior is unchanged, so the tests aren't altered. I'm not 100% sure if it's possible to test the "return bomb" case given that `receiveTeleporterMessage` doesn't have any return value